sympy/assumptions/ask.py

```diff
@@ -729,6 +729,44 @@ def is_true(self):
         """
         return Predicate('is_true')
 
+    # XXX: empty_matrix or just empty?
+    @predicate_memo
+    def empty_matrix(self):
+        """
+        A matrix having at least one dimension equal to zero.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, MatrixSymbol
+        >>> X = MatrixSymbol('X', 0, 0)
+        >>> Y = MatrixSymbol('Y', 2, 0)
+        >>> Z = MatrixSymbol('Z', 2, 2)
+        >>> ask(Q.empty_matrix(X))
+        True
+        >>> ask(Q.empty_matrix(Z*Y))
+        True
+        >>> ask(Q.empty_matrix(Z))
+        False
+        """
+        return Predicate('empty_matrix')
+
+    # XXX: is scalar_matrix predicat needed?
+    @predicate_memo
+    def scalar_matrix(self):
```

Comment:
If we want to define the predicate scalar_matrix, then I think that the proper definition would be that a scalar matrix is a square matrix that behaves like a scalar, i.e., it commutes with all other matrices of the same shape. This means that it is diagonal with all diagonal entries the same (commutative) scalar. In other words, it is a scalar multiple of a unit matrix.